src/handlers/request/tests/RequestHandlers.spec.ts

```diff
@@ -0,0 +1,62 @@
+import {RequestHandler} from '..';
+import handleRequest from '../RequestHandlers';
+
+describe('RequestHandlers', () => {
+    it('returns provided init options if there are no handlers', () => {
+        const mockRequest = {
+            method: 'POST',
+            body: JSON.stringify('A random body'),
+            headers: new Headers({
+                'Content-Type': 'application/json',
+            }),
+        };
+
+        const requestInit = handleRequest(mockRequest);
+
+        expect(requestInit).toEqual({
+            method: 'POST',
+            body: JSON.stringify('A random body'),
+            headers: new Headers({
+                'Content-Type': 'application/json',
+            }),
+        });
+    });
+
+    it('returns modified init options according to the request handlers', () => {
+        const mockHandler1 = {
+            canProcess: () => true,
+            process: (request: RequestInit) => ({
+                ...request,
+                body: JSON.stringify(`This handler clears and sets a new body`),
+            }),
+        } as RequestHandler;
+
+        const mockHandler2 = {
+            canProcess: () => true,
+            process: (request: RequestInit) => ({
+                ...request,
+                headers: new Headers({
+                    'Content-Type': 'text/xml',
+                }),
+            }),
+        } as RequestHandler;
+
+        const mockRequest = {
+            method: 'POST',
+            body: JSON.stringify('A random body'),
+            headers: new Headers({
+                'Content-Type': 'application/json',
+            }),
+        } as RequestInit;
+
+        const requestInit = handleRequest(mockRequest, [mockHandler1, mockHandler2]);
```

Comment:
You could add a test with an empty array for your second parameter, and we should get the same result as your first test.